client/extensions/woocommerce/state/sites/promotions/helpers.js

```diff
@@ -8,19 +8,32 @@ export function createPromotionFromProduct( product ) {
 	const startDate = product.date_on_sale_from || undefined;
 	const endDate = product.date_on_sale_to || undefined;
 	const productId = product.id;
+	const parentId = product.productId;
 
 	return {
 		id: 'p' + product.id,
-		name: product.name,
+		name: createProductName( product ),
 		type: 'product_sale',
 		appliesTo: { productIds: [ product.id ] },
 		salePrice,
 		startDate,
 		endDate,
 		productId,
+		parentId,
 	};
 }
 
+function createProductName( product ) {
+	if ( product.productId ) {
+		// It's a variation, append the options.
+		const options = product.attributes.map( attr => attr.option ).join( ', ' );
```

Comment:
You could use [`formattedVariationName `](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/lib/formatted-variation-name/index.js#L10) here, basically does the same thing but then we'd be consistent across Store.

Reply:
Oh, thanks!